### PR TITLE
Fix override of existing values in prepend & append

### PR DESCRIPTION
**Pull request by nicolasbender**
## Summary
Currently, when using `NewFormattedMapFromEnvironment(environment)`, entries like "$KEY:value" are overwritten by the last loop execution when using both append and prepend on the same $KEY. This fix appends and prepends values to entries which are already existing.

## Use Cases
We stumbled upon this in the following PR: https://github.com/paketo-buildpacks/npm-install/pull/688#discussion_r1651028653. Integration tests in this project are not working because they check for specific log entries written with `NewFormattedMapFromEnvironment(environment)`, which yield the wrong result, because they overwrite "$KEY:value" when append and prepend are used on the same $KEY.

## Checklist

* [ ] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: scribe/formatted_map.go
Comment (c0d1ngm0nk3y) [inline scribe/formatted_map.go]:
```suggestion
			existingValue, ok := envMap[parts[0]]
			if !ok {
				existingValue = "$" + parts[0]
			}
			envMap[parts[0]] = strings.Join([]string{value, fmt.Sprintf("%v", existingValue)}, environment[parts[0]+".delim"])
```

File: scribe/formatted_map.go
Comment (c0d1ngm0nk3y) [inline scribe/formatted_map.go]:
```suggestion
			existingValue, ok := envMap[parts[0]]
			if !ok {
				existingValue = "$" + parts[0]
			}
			envMap[parts[0]] = strings.Join([]string{fmt.Sprintf("%v", existingValue), value}, environment[parts[0]+".delim"])
```

**Review by c0d1ngm0nk3y — COMMENTED** [review]
Some suggestions. Mostly a matter of taste, but `existingValue.(string)` might cause a panic.

**Comment by robdimsdale** [comment]
Hello and welcome, @nicolasbender !

This broadly looks good - thanks for adding test coverage - and I'd like to see this PR incorporate the changes suggested by @c0d1ngm0nk3y . I don't want to just merge them in myself in case you have opinions on them.

**Comment by ForestEckhardt** [comment]
@nicolasbender To be clear. This will only affect the way these environment variables are ultimately logged no functionality will change as the setting of environment variables from the files generated is done by the lifecycle. I think that this is still a great fix I just wanted to lay that out cause it is unclear to me in the write up that is the intended outcome.

**Review by ForestEckhardt — APPROVED** [review]

**Comment by nicolasbender** [comment]
@ForestEckhardt 
> This will only affect the way these environment variables are ultimately logged

Yes, we are aware of that. The issue we had with this was only in some integration tests from the npm-install buildpack which tested the outcome of the logged result and not the result itself (https://github.com/paketo-buildpacks/npm-install/pull/688#discussion_r1651028653). And since the logging did not work properly, the integration tests failed.

**Comment by ForestEckhardt** [comment]
Perfect!

**Review by robdimsdale — APPROVED** [review]

**Review by c0d1ngm0nk3y — APPROVED** [review]

**Comment by nicolasbender** [comment]
Hi @ForestEckhardt @robdimsdale, thanks for merging! Is it possible to release this change in a patch version? Although this is a very small change, [this PR](https://github.com/paketo-buildpacks/npm-install/pull/688) is blocked until the next release of packit including this change.

**Comment by c0d1ngm0nk3y** [comment]
@paketo-buildpacks/tooling-maintainers Are there any plans on releasing `packit`?

**Comment by mhdawson** [comment]
@sophiewigmore not sure if you are involved in this repo, but getting a release would be great to unlock the Node.js pr.

**Pull request merged**
